packages/seacas/libraries/ioss/src/Ioss_ElementTopology.h

```diff
@@ -117,14 +122,20 @@ namespace Ioss {
     bool operator!=(const Ioss::ElementTopology &rhs) const;
     bool equal(const Ioss::ElementTopology &rhs) const;
 
+    virtual ElementPermutation *permutation() const;
+    virtual const std::string &base_topology_permutation_name() const;
   protected:
     ElementTopology(std::string type, std::string master_elem_name, bool delete_me = false);
+    virtual bool validate_permutation() const {return true;}
 
   private:
     bool              equal_(const Ioss::ElementTopology &rhs, bool quiet) const;
     const std::string name_;
     const std::string masterElementName_;
 
+    static Ioss::ElementShapeMap shapeToPermutationNameMap_;
```

Comment:
Or, these could just be in `namespace {` inside Ioss_ElementTopology.C can't they?  Not sure if it benefits having them defined here...

Reply:
Hmm, I suppose that's less clutter in `Ioss_ElementTopology` since they are really only used in one place...